### DISCUSSION TITLE
Worker running out of disk space

### DISCUSSION THREAD

**Discussion by maximveksler** [community/community#26357] [Actions]
<p>Hello Github Actions,</p>
<p>We are using Github Action to build our TensorFlow docker images. We’ve seen that some actions are failing on “out of disk space” error.</p>
<pre><code>#12 58.88 Downloading tensorflow-1.15.2-cp37-cp37m-manylinux2010_x86_64.whl (110.5 MB)
**#12 70.86 ERROR: Could not install packages due to an EnvironmentError: [Errno 28] No space left on device**#12 70.86
#12 ERROR: executor failed running [/bin/bash -c python3.7 -m pip install --upgrade pip &amp;&amp; python3.7 -m pip install -r requirements-gpu.txt]: runc did not terminate sucessfully
------
 &gt; [6/6] RUN --mount=type=ssh python3.7 -m pip install --upgrade pip &amp;&amp; python3.7 -m pip install -r requirements-gpu.txt:
------
failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: rpc error: code = Unknown desc = failed to build LLB: executor failed running [/bin/bash -c python3.7 -m pip install --upgrade pip &amp;&amp; python3.7 -m pip install -r requirements-gpu.txt]: runc did not terminate sucessfully
##[error]Process completed with exit code 1.
</code></pre>
<p>While debugging this we’ve discovred that the 20GB space provided by the build worker is mounted under /mnt while the docker daemon is configued to use /var/lib/docker on the root fs which has ~7GB of free space.</p>
<p>To workaround this issue we’ve added a docker configuartion step to have it use the /mnt folder instead of the the /var/lib disk location.</p>
<p>Pleae find here for the benefit of the community:</p>
<pre><code>steps:
    - name: Patch Docker Daemon data-root
      run: |
        DOCKER_DATA_ROOT='/mnt/var/lib/docker'
        DOCKER_DAEMON_JSON='/etc/docker/daemon.json'
        sudo mkdir -p "${DOCKER_DATA_ROOT}"
        jq --arg dataroot "${DOCKER_DATA_ROOT}" '. + {"data-root": $dataroot}' "${DOCKER_DAEMON_JSON}" &gt; "/tmp/docker.json.tmp"
        sudo mv "/tmp/docker.json.tmp" "${DOCKER_DAEMON_JSON}"
        sudo systemctl restart docker
</code></pre>
<p>I would be happy to have the worker by deafult set to use the /mnt path, or perhpas even better have the deafult “/” mount disk extended to include +20GB of storage data which would eliviate the need to configure this custom path (as it’s likely that other tools will use /var path as well reaching the same issue as described above.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/maximveksler">@maximveksler</a> ,</p>
<p>Many thanks for your sharing! It will be quite helpful for other guys who encounter same diskspace problem.</p>
<p>Regarding the default worker path, You can raise a feedbac.k ticket <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> where github product manager will take a review. Happy coding!</p>

**Comment by maximveksler** [comment]
<blockquote>
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> wrote:</p>
</blockquote>
<blockquote>
<p>You can raise a feedbac.k ticket <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> where github product manager will take a review. Happy coding!</p>
</blockquote>
<p>Yes thank you, that was done combined with posting the thread.</p>

**Comment by finalspy** [comment]
<p>Hi, I ran into the same kind of issue ie. “No Space Left On Device” using github actions for building several python based docker images (with spark and lot of libs…).</p>
<p>We need at least 60Gb of free space, may be more to ensure we can build all the docker images in a row. So above customisation isn’t enough for us.</p>
<p>So my question is simple, <strong>is there a way to get more space for our builds ?</strong></p>
<p>I know I could find ways to optimize our builds by purging temporary contents, pruning useless docker layers etc. but isn’t the goal of external CI/CD mechanism like actions to concentrate on our code not on the CI/CD ? If so we shouldn’t have to care about resources limits or allocation.</p>

**Comment by finalspy** [comment]
<p>Reply to myself in case others have the same issue : use external runners !</p>
<ul>
<li>[About] (<a href="https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners" rel="nofollow noopener">https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners</a>)</li>
<li>[Using] (<a href="https://docs.github.com/en/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow" rel="nofollow noopener">https://docs.github.com/en/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow</a>)</li>
<li>[Labels] (<a href="https://docs.github.com/en/actions/hosting-your-own-runners/using-labels-with-self-hosted-runners" rel="nofollow noopener">https://docs.github.com/en/actions/hosting-your-own-runners/using-labels-with-self-hosted-runners</a>)</li>
<li>[Workflow] (<a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on" rel="nofollow noopener">https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on</a>)</li>
</ul>

**Comment by maximveksler** [comment]
<p>I would not consider this a solution, as that means you are again maintaining infrastructure. In which case, you might as well consider using <a href="https://github.com/kubernetes/test-infra/tree/master/prow" rel="noopener nofollow ugc">Prow</a>.</p>
